src/overlays/actors/ovl_En_Mushi2/z_en_mushi2.c

```diff
@@ -30,134 +45,1192 @@ const ActorInit En_Mushi2_InitVars = {
+            TOUCH_NONE | TOUCH_SFX_NORMAL,
+            BUMP_NONE,
+            OCELEM_ON,
+        },
         { 0, { { 0, 0, 0 }, 5 }, 100 },
     },
 };
 
-// static ColliderJntSphInit sJntSphInit = {
-static ColliderJntSphInit D_80A6B974 = {
-    { COLTYPE_NONE, AT_NONE, AC_NONE, OC1_ON | OC1_TYPE_PLAYER | OC1_TYPE_1, OC2_TYPE_1, COLSHAPE_JNTSPH, },
-    1, D_80A6B950, // sJntSphElementsInit,
+static ColliderJntSphInit sJntSphInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_PLAYER | OC1_TYPE_1,
+        OC2_TYPE_1,
+        COLSHAPE_JNTSPH,
+    },
+    1,
+    sJntSphElementsInit,
+};
+
+static Color_RGBA8 D_80A6B984[] = {
+    { 170, 130, 90, 255 },
+    { 250, 250, 250, 255 },
+};
+
+static Color_RGBA8 D_80A6B98C[] = {
+    { 100, 60, 20, 255 },
+    { 180, 180, 180, 255 },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80A6BA04[] = {
+static s32 D_80A6B994 = 0;
+
+static EnMushi2Struct D_80A6B998[] = {
+    { 1, 0x1F40 },
+    { 2, 0xE0C0 },
+};
+
+static Vec3f D_80A6B9A0 = { 1.0f, 0.0f, 0.0f };
+static Vec3f D_80A6B9AC = { 0.0f, 1.0f, 0.0f };
+static Vec3f D_80A6B9B8 = { 0.0f, 0.0f, 1.0f };
+
+static MtxF D_80A6B9C4 = {
+    0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f,
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 10, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 700, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 20, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 20, ICHAIN_STOP),
 };
 
-#endif
+static f32 D_80A6BA14[] = { 0.06f, 0.1f, 0.13f };
+
+void func_80A687A0(EnMushi2* this) {
+    MtxF* matrix = SysMatrix_GetCurrentState();
+
+    matrix->mf[3][0] += this->unk_348 * this->unk_31C.x;
+    matrix->mf[3][1] += this->unk_348 * this->unk_31C.y;
+    matrix->mf[3][2] += this->unk_348 * this->unk_31C.z;
+}
+
+void func_80A68808(EnMushi2* this) {
+    SysMatrix_SetStateRotationAndTranslation(this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z,
+                                             &this->actor.shape.rot);
+    Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
+    Collider_UpdateSpheres(0, &this->collider);
+}
+
+s32 func_80A68860(EnMushi2* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 sp40;
+    CollisionPoly* sp3C;
+    f32 temp_f0 = func_800C411C(&globalCtx->colCtx, &sp3C, &sp40, &this->actor, &this->actor.world.pos);
+    s32 sp34;
+    f32 sp30;
+
+    if ((temp_f0 > (BGCHECK_Y_MIN + 1)) && ((this->actor.world.pos.y - 150.0f) < temp_f0)) {
+        return true;
+    }
+
+    return func_800CA1E8(globalCtx, &globalCtx->colCtx, this->actor.world.pos.x, this->actor.world.pos.z, &sp30, &sp34);
+}
+
+s32 func_80A68910(EnMushi2* this, GlobalContext* globalCtx) {
+    Actor* bean = globalCtx->actorCtx.actorList[ACTORCAT_ITEMACTION].first;
+    f32 minDist = SQ(100.0f);
```

Comment:
```suggestion
    f32 minDistSq = SQ(100.0f);
```
Would be slightly better